solara/util.py

```diff
@@ -257,9 +257,15 @@ def parse_timedelta(size: str) -> float:
         return float(size)
 
 
-def get_file_hash(path: Path, algorithm="md5") -> Tuple[bytes, str]:
+def get_file_hash(path: Union[Path, Generator], algorithm="md5") -> Tuple[bytes, str]:
     """Compute the hash of a file. Note that we also return the file content as bytes."""
-    data = path.read_bytes()
+    if isinstance(path, Generator):
+        data = b""
+        for file in path:
+            chunk = file.read_bytes()
+            data += chunk
+    else:
+        data = path.read_bytes()
```

Comment:
I had the same idea of accepting a Generator (can it be property typed? Generate[Path] orso?).
But it's more efficient to first create the hash object, and call .update on it, see
https://docs.python.org/3/library/hashlib.html#usage

Reply:
Yeah, from a MyPy error I got it seemed to be `Generator[Path, None, None]`.